valhalla/sif/dynamiccost.h

```diff
@@ -711,21 +738,32 @@ class DynamicCost {
             // If not, we should keep looking
 
             // We are in range at the time we are allowed at this edge
-            if (access_type == baldr::AccessType::kTimedAllowed)
+            if (access_type == baldr::AccessType::kTimedAllowed) {
+              destonly_access_restr_mask = tmp_mask;
               return true;
-            else if (access_type == baldr::AccessType::kDestinationAllowed)
+            } else if (access_type == baldr::AccessType::kDestinationAllowed)
               return allow_conditional_destination_ || is_dest;
             else
               return false;
           }
         }
       }
+
+      if (restriction.except_destination() &&
+          static_cast<size_t>(restriction.type()) < baldr::kAccessRestrictionMasks.size()) {
+        auto mask = baldr::kAccessRestrictionMasks[static_cast<size_t>(restriction.type())];
+        tmp_mask |= mask;
+        if ((destonly_access_restr_mask & mask) || allow_destination_only_)
+          continue;
+      }
+
       // In case there are additional restriction checks for a particular  mode,
       // check them now
       if (!ModeSpecificAllowed(restriction)) {
         return false;
       }
     }
+    destonly_access_restr_mask = tmp_mask;
```

Comment:
...and set `destonly_access_restr_mask` to the temp mask, so in the calling algorithm, this can be used to set the mask on the next label.